docs-src/typescript/how-to-run-a-worker-on-docker.md

```diff
@@ -9,29 +9,38 @@ tags:
   - typescript
 ---
 
+:::note
+
+Before proceeding, we recommend referring to our [production sample](https://github.com/temporalio/samples-typescript/tree/main/production),
+which shows how to save the compiled workflow bundle to a file and avoid compilation overhead on worker startup.
+
+:::
+
 Workers based on the TypeScript SDK can be deployed and run as Docker containers.
 
-At this moment, we recommend using Node.js 18.
-(Node.js 20 has known issues.)
-Both `amd64` and `arm64` platforms are supported.
+We recommend an LTS Node.js release such as 18 or 20.
+Both `amd64` and `arm64` architectures are supported.
 A glibc-based image is required; musl-based images are _not_ supported (see below).
 
-The easiest way to deploy a TypeScript SDK Worker on Docker is to start with the `node:18-bullseye` image.
+The easiest way to deploy a TypeScript SDK Worker on Docker is to start with the `node:20-bullseye` image.
 For example:
 
 ```dockerfile
-FROM node:18-bullseye
+FROM node:20-bullseye
 
+# For better cache utilization, copy package.json and lock file first and install the dependencies before copying the
+# rest of the application and building.
 COPY . /app
 WORKDIR /app
 
+# Alternatively, run npm ci, which installs only dependencies specified in the lock file and is generally faster.
```

Comment:
I think at this point, it should be reasonable to recommend this instead:

```
npm ci --omit dev
```
`--omit` was introduced in NPM 8, which itself got introduced in 16.11.0.

Reply:
`--omit dev` would require a separate build step, I'd avoid that for now.